### PR TITLE
CI add specific build job

### PR DESCRIPTION
**Pull request by SamYuan1990**
ref https://github.com/actions/checkout/tree/v3/#checkout-pull-request-head-commit-instead-of-merge-commit try to use github specific to check out code and meanwhile for container image tag.

### CODE REVIEW

**Comment by SamYuan1990** [comment]
as a try for #640

**Comment by SamYuan1990** [comment]
@rootfs , so far it require a full git commit SHA for example https://github.com/SamYuan1990/kepler/actions/runs/5004169317/jobs/8966392722 
https://github.com/SamYuan1990/kepler/actions/runs/5004190998/jobs/8966440203 (feature branch commit hash as `adf1106abc5d3dfb4bf9a227da7196c170c90fe9`)
any other suggestions?

File: .github/workflows/image_pr.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: imagePRlevel
+
+on:
+  workflow_dispatch:
+    # Inputs the workflow accepts.
+    inputs:
+      commitSHA:
+        description: "pr commit SHA"
+        required: true
```
Comment (rootfs) [inline .github/workflows/image_pr.yml:9]: do we have to provide sha manually or it can be set to the last commit?

Comment (SamYuan1990) [reply .github/workflows/image_pr.yml:9]: manually

**Review by rootfs — APPROVED** [review]

**Pull request merged**